client/src/containers/map/popup/item.tsx

```diff
@@ -28,42 +29,69 @@ const PopupItem = ({ id }: PopupItemProps) => {
     populate: 'metadata',
   });
 
+  const [featuresData, setfeaturesData] = useState<GeoJsonProperties | undefined>();
   const attributes = data?.data?.attributes as LayerTyped;
   const source = attributes.config.source;
   const click = attributes.interaction_config.events.find((ev) => ev.type === 'click');
+  const fetchLayer = attributes.interaction_config?.layer?.trim();
+  const url = attributes.interaction_config?.url?.trim();
 
-  const DATA = useMemo(() => {
-    if (source.type === 'vector' && rendered && popup && map) {
-      const point = map.project(popup.lngLat);
-
-      // check if the point is outside the canvas
-      if (
-        point.x < 0 ||
-        point.x > map.getCanvas().width ||
-        point.y < 0 ||
-        point.y > map.getCanvas().height
-      ) {
-        return DATA_REF.current;
+  useEffect(() => {
+    const fetchData = async () => {
+      if (!rendered || !popup || !map) return;
+      if (['raster', 'vector'].includes(source.type)) {
+        const point = map.project(popup.lngLat);
+        // check if the point is outside the canvas
+        if (
+          point.x < 0 ||
+          point.x > map.getCanvas().width ||
+          point.y < 0 ||
+          point.y > map.getCanvas().height
+        ) {
+          setfeaturesData(DATA_REF.current);
+          return;
+        }
       }
-      const query = map.queryRenderedFeatures(point, {
-        layers: layersInteractiveIds.map((id) => id.toString()),
-      });
-
-      const d = query.find((d) => {
-        return d.source === source.id;
-      })?.properties;
-
-      DATA_REF.current = d;
-
-      if (d) {
-        return DATA_REF.current;
+      if (source.type === 'raster') {
+        const point = map.project(popup.lngLat);
+        const bounds = map.getBounds();
+        const bbox = `${bounds.getWest()},${bounds.getSouth()},${bounds.getEast()},${bounds.getNorth()}`;
+        const width = map.getContainer().clientWidth;
+        const height = map.getContainer().clientHeight;
+        const x = Math.round(point.x);
+        const y = Math.round(point.y);
+        const response = await fetch(
+          // Projection should be included in the url: ?crs=EPSG:4326
+          `${url}&request=GetFeatureInfo&transparent=true&format=image/png&exceptions=application/vnd.ogc.se_xml&styles=&feature_count=10&service=WMS&version=1.1.1&layers=${fetchLayer}&QUERY_LAYERS=${fetchLayer}&x=${x}&y=${y}&bbox=${bbox}&width=${width}&height=${height}&info_format=application/json
+          `,
+        );
```

Comment:
We don't want to hard-code this. Instead, we could use params from `params_config` and inject specific values such as the coordinates and the bbox.

Reply:
Shouldn't we use `interaction_config` for this params instead of `params_config`?

Reply:
Ok, I've seen that for some compound layers you have to use `params_config` too